lib/messages.h

```diff
@@ -80,7 +81,7 @@ void msg_add_option_group(GOptionContext *ctx);
 
 #define msg_debug(desc, tag1, tags...) 						  \
 	do { 									  \
-	  if (G_UNLIKELY(debug_flag))             				  \
+	  if (G_UNLIKELY(debug_flag) || G_UNLIKELY(startup_debug_flag))             				  \
 	    msg_event_suppress_recursions_and_send(                               \
```

Comment:
I agree with @beng94 here: `syslog-ng -dr` turning off debugging after startup would be surprising. Mind you, using both would be uncommon: when running from a startup script or similar, you don't want both. When running by hand, you most likely want `-d`, and not `-r`.

I'd still opt for a separate flag, that causes the least amount of surprise.